src/app/_hooks/useFilter.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useMemo } from 'react'
+
+import { type NextServerSearchParams } from '@/types/searchParams'
+import { type Object } from '@/types/utils'
+
+import { ALL_FILTERS_ID } from '@/constants/filterConstants'
+
+import { normalizeQueryParam } from '@/utils/queryUtils'
+
+export type UseFilterProps<Entry extends Object> = {
+  searchParams: NextServerSearchParams
+  entries: Array<Entry>
+  filterKey: string
+  filterFn: (entry: Entry, query?: string) => boolean
+}
+
+export function useFilter<Entry extends Object>({
+  searchParams,
+  entries,
+  filterKey,
+  filterFn,
+}: UseFilterProps<Entry>) {
+  const filterQuery = normalizeQueryParam(searchParams, filterKey)
+
+  const filterByQuery = useCallback(
+    (entry: Entry) => filterFn(entry, filterQuery),
+    [filterFn, filterQuery],
+  )
+
+  const filteredResults = useMemo(() => {
```

Comment:
`filteredEntries`? - Can we look into the rest where we use entries vs. results just to make sure it still all makes sense.

Reply:
The other hooks use results instead of entries -> [Ticket](https://www.notion.so/filecoin/FF-Followup-Dev-Work-d6616163f43c4e7db78edf218cebcc8d?pvs=4#1537631f282580d98665c69c4e4b328d)